common-utils/src/common/deployment/__init__.py

```diff
@@ -497,27 +668,55 @@ def process_project_flows(self, validate_only: bool = False) -> None:
         Raises:
             Exception: Exception will thrown for bad FLOW_SPEC definitions.
         """
+        # account_id and ecs client is only populated on deployment
+        # need default values for validation to work and prevent unbound variables
+        account_id = ""
+        ecs_client = object()
         flows_path = self._prefect_flows_folder
+        # setup to aggregate errors for raising at the end as needed
         flow_errors = []
         missing_flow_specs = []
+        # this is the deployment path
         if not validate_only:
+            # command type used for DRY logging
             command_type = "deployment"
+            # only need AWS for deployment
+            # single account_id and ecs client for entire deployment workflow
+            from boto3 import session
+
+            init_session = session.Session()
+            account_id = get_aws_account_id(init_session)
+            ecs_client = init_session.client("ecs")
+
         else:
+            # set command type to validation to skip deployment
             command_type = "validation"
+
+        # loop through list of all files that end with "_flow.py"
         for name in glob.glob(os.path.join(flows_path, "**/*_flow.py"), recursive=True):
             path_object = Path(name)
             file_name = path_object.name
             mod = file_name.split(".")[0]
             LOGGER.info(f"Running {command_type} for flow: {mod} at path: {name}...")
             try:
+                # load FLOW_SPEC global from file
                 x = SourceFileLoader(mod, name).load_module().FLOW_SPEC
+                # validate docker env is registered in pyproject.toml
+                if x.docker_env not in self._pyproject_metadata.docker_envs.keys():
+                    raise ValueError(
+                        "Docker env does not exist in pyproject.toml.  It must be the key name in a '[tool.prefect.envs.<key name>]' configuration."
+                    )
+                # only run deployment when validate_only is False
```

Comment:
moving the docker env validation to `deploy-cli` call so that at least a bad env will fail on PR